File: play.py
#!/usr/bin/python
import os
import sys
import glob

USER=os.getlogin()
MUSIC_DIR=os.path.join('/Users',USER,'Music/iTunes/iTunes Media/Music/')
PLS=os.path.join('/Users',USER,'Music/playlist.pls')

def my_list_dir(somedir):
    return [elm for elm in os.listdir(somedir) if not elm.startswith('.')]

arg=' '.join(sys.argv[1:])
band, album = arg.split('/', 1) if arg.count('/') > 0 else arg, None

bands = my_list_dir(MUSIC_DIR)

tracks = []
for bnd in bands:
    if bnd.lower().startswith(band.lower()):
        band_dir=os.path.join(MUSIC_DIR, bnd)
        band_alb=my_list_dir(band_dir)
        for alb in band_alb:
            if not album or alb.lower().startswith(album.lower()):
                alb_dir=os.path.join(band_dir, alb, '*.mp3')
                tracks.extend(glob.glob(alb_dir))

_pls = "\n".join(tracks)
if len(_pls) > 0:
    pls = open(PLS, 'w')
    pls.write(_pls)
    pls.close()
    os.system('mpg123 -C -@ ' + PLS)
                

File: play.php
#!/usr/bin/php
<?php
define('MUSIC_DIR', '/Users/dorofey/Music/iTunes/iTunes Media/Music/');
$dh = opendir(MUSIC_DIR);
while (($file = readdir($dh)) !== false) {
    if(strpos($file, '.') !== 0) $bands[] = $file;
}
closedir($dh);

$_ar = $argv;
array_shift($_ar);
$_ar = explode('/',implode(' ', $_ar));

$_band = $_ar[0];
$_album= (isset($_ar[1]))?$_ar[1]:'';
foreach ($bands as $band) {
    if(strpos(strtolower($band), strtolower($_band)) === 0) {
        printf("Band: %s found!\n", $band,$_band,$_album);
        $banddir = MUSIC_DIR . $band . '/';
        $dh = opendir($banddir);
        $albums = array();
        while (($file = readdir($dh)) !== false) {
            if(strpos($file, '.') !== 0 && (!$_album || strpos(strtolower($file), strtolower($_album)) === 0))
                $albums[] = $file;
        }
        closedir($dh);
        foreach($albums as $album) {
            $dir = $banddir . $album . '/';
            $dh = opendir($dir);
            while (($file = readdir($dh)) !== false) {
                if(strpos($file, '.') !== 0)
                    $tracks[] = $dir . $file;
            }
            
        }
        if(count($tracks > 0)) {
            $fp = fopen('/Users/dorofey/Music/playlist.txt', 'w');
            foreach($tracks as $track) {
                fputs($fp, $track . "\n");
            }
            fclose($fp);
            exec('mpg123 -C -@ ' . '/Users/dorofey/Music/playlist.txt');
        }
    }
}

?>